urllib3/util/ssl_.py

```diff
@@ -53,6 +54,12 @@ def _const_compare_digest_backport(a, b):
     OP_NO_SSLv2, OP_NO_SSLv3 = 0x1000000, 0x2000000
     OP_NO_COMPRESSION = 0x20000
 
+try:
+    import ipaddress
```

Comment:
You can use the socket module to check if a string is an IP address:

```
import socket

def check_ipaddr(hostname):
    families = [socket.AF_INET]
    if hasattr(socket, 'AF_INET6'):
        families.append(socket.AF_INET6)
    for af in families:
        try:
            socket.inet_pton(af, hostname)
        except socket.error:
            pass
        else:
            return True
    return False
```

Reply:
Nice- that's clever. We'd actually been kicking around ways to do this in Ansible without taking the ipaddress dep- too bad it doesn't always work for IPv6 on Py2, but still...

Reply:
It should always work when the platform support IPv6. Needs verification :)